src/go/k8s/pkg/resources/resource.go

```diff
@@ -22,6 +22,8 @@ import (
 
 // Resource decompose the reconciliation loop to specific kubernetes objects
 type Resource interface {
+	Reconciler
```

Comment:
On the `commit body` front. the commit does not tell the reviewer why introducing a reconciler interface is useful or necessary.

Reply:
good point!